lua/starfall/libs_sh/holograms.lua

```diff
@@ -331,46 +371,45 @@ else
 			holo:DisableMatrix("RenderMultiply")
 		end
 	end
-
+	
 	--- Parents a hologram
-	-- @param Entity? ent Entity parent (nil to unparent)
-	-- @param number? attachment Optional attachment ID
-	function hologram_methods:setParent(ent, attachment)
-
+	-- @param Entity? parent Entity parent (nil to unparent)
+	-- @param number|string? attachment Optional attachment name or ID
+	-- @param boolean? bone True to parent to a bone instead of an attachment (`attachment` parameter must be a number)
+	function hologram_methods:setParent(parent, attachment, bone)
 		local holo = getholo(self)
-
 		checkpermission(instance, holo, "hologram.setParent")
-
-		if ent ~= nil then
-			local parent = getent(ent)
-
-			if attachment == nil then attachment = -1 end
-			checkluatype(attachment, TYPE_NUMBER)
-
-			if not parent.sf_children then
-				parent.sf_children = {}
+		
+		if parent ~= nil then
+			parent = getent(parent)
+			if bone ~= nil then checkluatype(bone, TYPE_BOOL) end
+			if bone then
+				attachment = attachment or 0
+				checkluatype(attachment, TYPE_NUMBER)
+			else
+				if isstring(attachment) then
+					attachment = parent:LookupAttachment(attachment)
+				else
+					attachment = attachment or -1
+					checkluatype(attachment, TYPE_NUMBER)
+				end
 			end
-
-			if holo.sf_parent then
-				holo.sf_parent.sf_children[holo] = nil
+			
+			local parentFunc = bone and "FollowBone" or "SetParent"
```

Comment:
Instead of a string, you can just give it the function, `holo.FollowBone` `holo.SetParent`

Reply:
Then you just call it with `data.func(child, parent, data.bone)`

Reply:
It'd still be `data.func(holo, child, parent, data.bone)`, but the reason I'm not storing functions directly is to allow monkey-patching. Will change it tho, sure.

Reply:
I guess it's fine as is